engine/runtime/src/main/java/org/enso/interpreter/node/callable/IndirectInvokeCallableNode.java

```diff
@@ -52,6 +52,37 @@ public abstract Object execute(
       InvokeCallableNode.ArgumentsExecutionMode argumentsExecutionMode,
       BaseNode.TailStatus isTail);
 
+  @Specialization
+  Object invokeWithWarnings(
+          WithWarnings warnings,
```

Comment:
Rather than expecting a value of type `WithWarnings` you should specialize on `Object` with a guard `warnings.Library.hasWarnings(obj)`.
The result is the same but I believe it's less likely to cause bugs in the future.